### DISCUSSION TITLE
There is a difference between normal links and the link to an attachement

### DISCUSSION THREAD

**Discussion by pmottishaw** [akosbalasko/yarle#307] [Ideas]
It would be very useful if Yarle had a setting to indicate the difference between a normal link and a link to an attachment. The link to the attachment in Evernote is effectively private to the note that contains the attachment ... no other note will ever link to the attachment, so Yarle can safely change the name of the attachment and the link to it. I would suggest a setting that allows a prefix such as "attach-" to be set. The problem below is an example of why this is useful.

I had a big problem with the Obsidian notes that Yarle generated that was the result of using "import folders" in Evernote. This is a feature of Evernote where you can ask it to watch a folder and if a new file is saved in the folder Evernote will create a new note with that file as an attachment. The Yarle problem occurs because **the new note created by Evernote has the same name as the attachment including the extension**. As a consequence, when Yarle imports this it creates a note and attachment with the same name. The links to the original note seem to randomly point to either the attachment or the note. After a while I realised that Yarle truncates some attachment names ( longer than some number of characters?) and in this case there is no problem. In all other cases Obsidian seemed to link to the attachment not the note - and this causes lots of problems.

I found a work around of setting the Yarle config to include the .md extension in links which seemed to fix the problem, but it would seem best to make the difference between the attachment and the note clear by prepending something (attachment- or just _) to the attachment.

It would also be good to have control over the truncation of attachment file names ... I would like them to be much longer.

**Comment by akosbalasko** [comment]
Hi @pmottishaw !

Ahh.. I've got your point, and totally agree with it. So let me create two issues to cover these features: 
- add prefix to attachments (it could be specified as a free text in the UI)
- set the truncation length of the file names (up to the limit what the target OS allows)

I'll keep you posted on the progress. 
Thank you for raising this up, cool requests!
